src/history/hs.c

```diff
@@ -1808,34 +1810,48 @@ __verify_history_store_id(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, uint32
 int
 __wt_history_store_verify_one(WT_SESSION_IMPL *session)
 {
-    WT_CURSOR *cursor;
-    WT_CURSOR_BTREE cbt;
+    WT_CURSOR *ds_cursor, *hs_cursor;
     WT_DECL_RET;
     WT_ITEM hs_key;
     uint32_t btree_id;
     int exact;
+    char *uri_data;
+    const char *open_cursor_cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL};
 
-    cursor = session->hs_cursor;
+    hs_cursor = session->hs_cursor;
     btree_id = S2BT(session)->id;
 
     /*
      * We are required to position the history store cursor. Set it to the first record of our btree
      * in the history store.
      */
     memset(&hs_key, 0, sizeof(hs_key));
-    cursor->set_key(cursor, btree_id, &hs_key, 0, 0);
-    ret = __wt_hs_cursor_search_near(session, cursor, &exact);
+    hs_cursor->set_key(hs_cursor, btree_id, &hs_key, 0, 0);
+    ret = __wt_hs_cursor_search_near(session, hs_cursor, &exact);
     if (ret == 0 && exact < 0)
-        ret = __wt_hs_cursor_next(session, cursor);
+        ret = __wt_hs_cursor_next(session, hs_cursor);
 
-    /* If we positioned the cursor there is something to verify. */
-    if (ret == 0) {
-        __wt_btcur_init(session, &cbt);
-        __wt_btcur_open(&cbt);
-        ret = __verify_history_store_id(session, &cbt, btree_id);
-        WT_TRET(__wt_btcur_close(&cbt, false));
+    if (ret == WT_NOTFOUND)
+        return (0);
+
+    WT_RET(ret);
+
+    if ((ret = __wt_metadata_btree_id_to_uri(session, btree_id, &uri_data)) != 0) {
+        F_SET(S2C(session), WT_CONN_DATA_CORRUPTION);
+        WT_ERR_PANIC(
+          session, WT_PANIC, "Unable to find btree id %" PRIu32 " in the metadata file", btree_id);
     }
-    return (ret == WT_NOTFOUND ? 0 : ret);
+
+    /* If we positioned the cursor there is something to verify. */
+    WT_ERR(__wt_open_cursor(session, uri_data, NULL, open_cursor_cfg, &ds_cursor));
```

Comment:
@agorrod This call returns ebusy because the btree has WT_BTREE_VERIFY flag set which is one of the flags classified as WT_BTREE_SPECIAL_FLAGS.
```
    /*
     * If this session already has exclusive access to the handle, there is no point trying to lock
     * it again.
     *
     * This should only happen if a checkpoint handle is locked multiple times during a checkpoint
     * operation, or the handle is already open without any special flags. In particular, it must
     * fail if attempting to checkpoint a handle opened for a bulk load, even in the same session.
     */
    if (dhandle->excl_session == session) {
        if (!LF_ISSET(WT_DHANDLE_LOCK_ONLY) &&
          (!F_ISSET(dhandle, WT_DHANDLE_OPEN) ||
            (btree != NULL && F_ISSET(btree, WT_BTREE_SPECIAL_FLAGS))))
                return (__wt_set_return(session, EBUSY));
            
        ++dhandle->excl_ref;
        return (0);
    }
```
I think the previous usage of `__wt_btcur_open ` is to work around this limitation. Do you know another way to work around this?